### ISSUE TITLE
`help(1)` crashes the interpreter

### ISSUE THREAD

**Issue by Fidget-Spinner** [pylbbv/pylbbv#15]
# Crash report

`help(1)` crashes the interpreter due to operand stack overrun.

**Comment by JuliaPoo** [comment]
## EDIT: Explanation for weird log:

This might not be the reason for the error, since `LOAD_CONST` in this case is a special case in `_PyTier2_Code_DetectAndEmitBB` and handled here:
```c
// We break our own rules for more efficient code here.
// NOTE: THIS MODIFIES THE TYPE CONTEXT.
if (_PyLong_IsNonNegativeCompact((PyLongObject *)GET_CONST(oparg))) {
    write_i = emit_i(write_i, LOAD_CONST, curr->op.arg);

    // Type propagate
    _PyTier2TypeContext *type_context = starting_type_context;
    _Py_TYPENODE_t **type_stackptr = &type_context->type_stack_ptr;
    *type_stackptr += 1;
    TYPE_OVERWRITE((_Py_TYPENODE_t *)_Py_TYPENODE_MAKE_ROOT((_Py_TYPENODE_t)&PySmallInt_Type), TYPESTACK_PEEK(1), true);
    continue;
}
```

Weird log:
```
  [*] Copying type context
      Stack: 000001E642892CA0: [ ? ?.? ? ? ?]
      Locals 000001E642894EE0: [ ? ? ? ? ? ? ? ? ?]
offset: 477
  [-] Type stack bef: 2
  [-] Type propagating across: LOAD_GLOBAL : 11
  [-] Type stack aft: 4
      Stack: 000001E642898220: [ ? ? ? ?.? ?]
      Locals 000001E642895300: [ ? ? ? ? ? ? ? ? ?]
offset: 482
  [-] Type stack bef: 4
  [-] Type propagating across: LOAD_FAST : 1
  [-] Type stack aft: 5
      Stack: 000001E642898220: [ ? ? ? ? ?->locals[1].?]
      Locals 000001E642895300: [ ? ? ? ? ? ? ? ? ?]
offset: 483
  [-] Type stack bef: 5
  [-] Type propagating across: LOAD_GLOBAL : 6
  [-] Type stack aft: 6
      Stack: 000001E642898220: [ ? ? ? ? ?->locals[1] ?]
      Locals 000001E642895300: [ ? ? ? ? ? ? ? ? ?]
offset: 488
offset: 489
emitted type guard 000001E6424C12D6 CHECK_FLOAT
Creating a BB Metadata with ID 44
Generated BB T2 Start: 000001E6424C12BE, T1 offset: 489
  [+] Executing: Stack:    2; Offset:  423; LOAD_GLOBAL_BUILTIN:11;
  [+] Executing: Stack:    4; Offset:  428; LOAD_FAST:1;
  [+] Executing: Stack:    5; Offset:  429; LOAD_GLOBAL_MODULE:6;
  [+] Executing: Stack:    6; Offset:  434; LOAD_CONST:2; <-- This was never type propagated across?? Otherwise the stack would have blown
  ```
  
 Reproduce:
  
 In `ceval_macros.h` replace `TARGET` with this:
```c
  #  define TARGET(op) case op: TARGET_##op: if (frame->f_code->_tier2_info != NULL) {fprintf(stderr, "  [+] Executing: Stack: %4d; Offset: %4d; %s:%d;\n", STACK_LEVEL(), INSTR_OFFSET(), #op, oparg);}\
        ; INSTRUCTION_START(op);
```

Run `help(1)` in the interpreter.

**Comment by JuliaPoo** [comment]
Following the fixes in #19 and #21, `help(1)` now crashes at
```py
>>> help(1)
Assertion failed: next_instr + INLINE_CACHE_ENTRIES_BB_BRANCH == t2_nextinstr, file C:\Users\nailu\Documents\Projects\CPython-Tier-2\pylbbv\Python\generated_cases.c.h, line 4295
```
This crash could be related to #14.
